internal/feed/reddit.go

```diff
@@ -28,8 +28,12 @@ type subredditResponseJson struct {
 	} `json:"data"`
 }
 
-func FetchSubredditPosts(subreddit string) (ForumPosts, error) {
-	requestUrl := fmt.Sprintf("https://www.reddit.com/r/%s/hot.json", url.QueryEscape(subreddit))
+func FetchSubredditPosts(subreddit string, httpProxy string) (ForumPosts, error) {
+	subreddit = url.QueryEscape(subreddit)
+	requestUrl := fmt.Sprintf("https://www.reddit.com/r/%s/hot.json", subreddit)
+	if httpProxy != "" {
+		requestUrl = fmt.Sprintf(httpProxy, subreddit)
```

Comment:
Could we maybe change this to something like:

```go
strings.Replace(httpProxy, "{REQUEST-URL}", requestUrl)
```

It helps with making `%s` less ambiguous and also gives us more control over the full URL since I'm planning on adding the ability to sort by new, top, etc and passing just the subreddit won't suffice.